File: 4.system_calls.py
import os
import subprocess

def main():
    # Forking a child process
    child_pid = os.fork()
    
    if child_pid == 0:
        print("Child Process: PID =", os.getpid())
        os.execlp('ls', 'ls', '-l')
    else:
        print("Parent Process: PID =", os.getpid())
        os.wait()
        print("Child process has finished execution.")

if __name__ == "__main__":
    main()

File: 4.system_calls.c
#include <stdio.h>
#include <unistd.h>
#include <sys/wait.h>

int main() {
    pid_t child_pid;

    // Forking a child process
    child_pid = fork();

    if (child_pid == 0) {
        printf("Child Process: PID = %d\n", getpid());
        execlp("ls", "ls", "-l", NULL);
    } else {
        printf("Parent Process: PID = %d\n", getpid());
        wait(NULL);
        printf("Child process has finished execution.\n");
    }

    return 0;
}